### DISCUSSION TITLE
How can you validate a method that only allows POST or GET or DELETE, ETC http methods, on the server?

### DISCUSSION THREAD

**Discussion by fchinch** [serverpod/serverpod#881] [Q&A]
How can you validate a http method that only allows POST or GET or DELETE, ETC , on the server?

**Comment by fchinch** [comment]
Validate it this way:

```

 Future<bool> addArticle(Session session, Article article) async {
    session as MethodCallSession;
    if (session.httpRequest.method != 'POST') {
      return false;
    }

    try {
      await Article.insert(session, article);
      return true;
    } catch (e) {
      return false;
    }
  }

**Comment by vlidholt** [comment]
Serverpod always uses http `POST`. If you want to write custom HTTP responses, you can use the Relic webserver, some more info here:
https://docs.serverpod.dev/concepts/webserver

**Comment by fchinch** [comment]
Hi @vlidholt , I was testing the endpoint in Postman, I realized that the method I was implementing also allows me to send the information as GET. 

So I wanted to limit to only use POST.

Basically I test everything with curl/postman/insomnia , if I provide an endpoint to a third party , I want to have these validations.
